### PR TITLE
Validate numeric values in the Settings model.

### PR DESCRIPTION
**Pull request by osmanzeki**
A simple addition that displays an error message to the user when non numeric values are sent in the Settings form for the authorization code.

### CODE REVIEW

File: models/Settings.php [hunk 1]
```diff
@@ -10,11 +10,24 @@ class Settings extends Model
...
-        'digits'                => 'required_if:is_enabled,1,true',
+        'digits' => 'required_if:is_enabled,1,true|numeric',
     ];
     // A unique code
     public $settingsCode = 'vannut_wipprotect_settings';
 
     // Reference to field configuration
     public $settingsFields = 'fields.yaml';
+
+    public function afterValidate()
```
Comment (osmanzeki) [inline models/Settings.php]: I'm not exactly sure if I should have used `beforeValidate` or `afterValidate` here. Both seem to work the same. I believe on is to do a custom validation before what is found in the `$rules` property is executed and the other is after. As long a an exception is returned, the data will be rejected either way.

Comment (vannut) [reply models/Settings.php]: Thanks @osmanzeki for the PRs!! 

AFAIK you don’t need to use an onBefore or onAfterValidate just to validate the user input. You have to initiate it in the yaml of the model. Ill look into it asap; and review your other PRs asap

Comment (osmanzeki) [reply models/Settings.php]: I tried using the "numeric" validation rule from Laravel, but it accepts hexadecimal values. If I don't do the afterValidate, the form does not show any error to the user but the password is also not saved (the value 0 is saved instead).

**Comment by vannut** [comment]
@osmanzeki Hi There! I just merged some of your PR's. But I was doubting about this one. As it complicates the plugin; and I was doubting why not to allow non-numeric `passwords`.

What are your thoughts about this?

**Comment by osmanzeki** [comment]
Hey, non-numeric passwords make sense too in my opinion. Shopify has this feature and allows for text-based passwords which works fine.

My only issue here was that if you enter a non-numeric password at the moment and press save, the value "0" is stores so the password does not work. If we change the behaviour to store text passcodes, it would also fix this behaviour so I am all for it. :)

**Comment by vannut** [comment]
All right. Then i'll ignore this PR ;)

**Pull request closed without merging**